src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/MessageBody.cs

```diff
@@ -273,6 +274,22 @@ private void TryStartTimingReads()
             }
         }
 
+        private void TryPauseTimingReads()
+        {
+            if (!RequestUpgrade)
```

Comment:
Was just asking. At the moment, we don't support upgradable requests with request bodies so this should be fine.

Btw, it would be cleaner to put this in the `ForUpgrade` class.